scripts/bump_sui_testnet_version.sh

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env bash
+# Copyright (c) Walrus Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+# This script creates a PR branch and updates Sui testnet versions in
+# selected files.
+
+set -Eeuo pipefail
+
+# Check required params.
+if [[ -z ${1:-} || $# -ne 1 ]]; then
+  echo "USAGE: bump_sui_testnet_version.sh <new-tag>"
+  exit 1
+else
+  NEW_TAG="$1"
+fi
+
+# (Loose) sanity check on tag format.
+if [[ ! "$NEW_TAG" =~ ^testnet-v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+  echo "Warning: NEW_TAG '$NEW_TAG' doesn't look like testnet-vX.Y.Z" >&2
+fi
+
+# Make sure GITHUB_ACTOR is set.
+if [[ -z "${GITHUB_ACTOR:-}" ]]; then
+  GITHUB_ACTOR="$(git config user.name 2>/dev/null || echo github-actions[bot])"
+fi
+
+# Set up branch for changes.
+STAMP="$(date +%Y%m%d%H%M%S)"
+BRANCH="${GITHUB_ACTOR}/bump-sui-${NEW_TAG}-${STAMP}"
+git checkout -b "$BRANCH"
+
+# Allow recursive globs.
+shopt -s globstar nullglob
+
+# List of relevant TOML locations (globs allowed).
+FILES=(
+  "contracts/**/Move.toml"
+  "docker/walrus-antithesis/sui_version.toml"
+  "Cargo.toml"
+  "testnet-contracts/**/Move.toml"
+)
+
+# Expand patterns into actual file paths.
+TARGETS=()
+for pat in "${FILES[@]}"; do
+  for f in $pat; do
+    [[ -f "$f" ]] && TARGETS+=("$f")
+  done
+done
+
+# Check if we found any targets.
+if [[ ${#TARGETS[@]} -eq 0 ]]; then
+  echo "No matching files found for update."
+  exit 0
+else
+  echo "Updating testnet tags in:"
+  printf '  - %s\n' "${TARGETS[@]}"
+
+  for f in "${TARGETS[@]}"; do
+    sed -i -E \
+      "s/( = \")testnet-v[0-9]+\.[0-9]+\.[0-9]+/\1${NEW_TAG}/g" "$f"
+  done
+fi
+
+# Update Cargo.lock
+echo "Running cargo build --release ..."
+cargo build --release
```

Comment:
Avoid failing the script if this fails.

```suggestion
cargo build --release ||:
```

Reply:
cargo failures exit with `1` so this will always get caught